tests/Functional/ContentOrganisationRenderTest.php

```diff
@@ -0,0 +1,209 @@
+    \Drupal::service('theme_installer')->install(['oe_theme']);
+    \Drupal::configFactory()
+      ->getEditable('system.theme')
+      ->set('default', 'oe_theme')
+      ->save();
+
+    // Rebuild the ui_pattern definitions to collect the ones provided by
+    // oe_theme itself.
+    \Drupal::service('plugin.manager.ui_patterns')->clearCachedDefinitions();
+
+    // Give anonymous users permission to view entities.
+    Role::load(RoleInterface::ANONYMOUS_ID)
+      ->grantPermission('view published skos concept entities')
+      ->grantPermission('view published oe_contact')
+      ->save();
+  }
+
+  /**
+   * Tests that the Organisation page renders correctly.
+   */
+  public function testOrganisationRendering(): void {
+    $file = file_save_data(file_get_contents(drupal_get_path('theme', 'oe_theme') . '/tests/fixtures/example_1.jpeg'), 'public://example_1.jpeg');
+    $file->setPermanent();
+    $file->save();
+
+    $media = $this->getStorage('media')->create([
+      'bundle' => 'image',
+      'name' => 'test image',
+      'oe_media_image' => [
+        'target_id' => $file->id(),
+        'alt' => 'Alt',
+      ],
+      'uid' => 0,
+      'status' => 1,
+    ]);
+    $media->save();
+
+    $contact = $this->getStorage('oe_contact')->create([
+      'bundle' => 'oe_general',
+      'name' => 'General contact',
+      'oe_email' => 'example@test.com',
+      'oe_phone' => '0123456789',
+      'oe_address' => [
+        'country_code' => 'BE',
+        'locality' => 'Brussels',
+        'address_line1' => 'My address',
+        'postal_code' => '1001',
+      ],
+    ]);
+
+    /** @var \Drupal\node\Entity\Node $node */
+    $node = $this->getStorage('node')->create([
+      'type' => 'oe_organisation',
+      'title' => 'My node title',
+      'oe_summary' => 'My introduction',
+      'body' => 'My body text',
+      'oe_organisation_acronym' => 'My acronym',
+      'oe_organisation_logo' => [
+        'target_id' => $media->id(),
+      ],
+      'oe_organisation_contact' => [$contact],
+      'oe_teaser' => 'The teaser text',
+      'oe_content_content_owner' => 'http://publications.europa.eu/resource/authority/corporate-body/COMMU',
+      'uid' => 0,
+      'status' => 1,
+    ]);
+    $node->save();
+    $this->drupalGet($node->toUrl());
+
+    // Assert page header - metadata.
+    $this->assertSession()->elementTextContains('css', '.ecl-page-header-core .ecl-page-header-core__meta', 'My acronym');
+    $this->assertSession()->elementTextContains('css', '.ecl-page-header-core h1.ecl-page-header-core__title', 'My node title');
+    $this->assertSession()->elementTextContains('css', '.ecl-page-header-core .ecl-page-header-core__description', 'My introduction');
+
+    // Assert navigation part.
+    $wrapper = $this->assertSession()->elementExists('css', '.ecl-row.ecl-u-mt-l');
+    $navigation = $this->assertSession()->elementExists('css', 'nav.ecl-inpage-navigation', $wrapper);
```

Comment:
Assert class can be created for In-page navigation template modules/oe_theme_helper/templates/oe-theme-helper-in-page-navigation.html.twig, but it isn't pattern, it is template. Anyway I think the same approach as with PatternAsserts can be implemented here.

Reply:
If its not a pattern I would not put it in the same location though, lets discuss

Reply:
done